src/wallet/rpcwallet.cpp

```diff
@@ -4793,7 +4793,13 @@ static std::optional<Memo> ParseMemo(const UniValue& memoValue)
                         assert(false);
                 }
             },
-            [](Memo result) { return result; }
+            [](Memo result) -> std::optional<Memo> {
+                if (result.ToBytes() == Memo::NoMemo().ToBytes()) {
+                    return std::nullopt;
+                } else {
+                    return result;
```

Comment:
```suggestion
                    return {result};
```
This is equivalent because there's a converting constructor from `T` to `std::optional<T>` (the `template <class U = T> constexpr optional( U&& value );` one that is listed last [here](https://en.cppreference.com/w/cpp/utility/optional/optional)), but as a matter of style I prefer we write it with the initializer-list braces. Non-blocking.